packages/shared/routes/dashboard/staking/views/StakingInfo.svelte

```diff
@@ -33,34 +44,51 @@
     $: $selectedAccountParticipationOverview, stakingEventState, setText()
     $: isAssemblyStaked, isShimmerStaked, stakingEventState, $selectedAccountId, setAnimation()
 
-    enum AnimationFileNumber {
-        NoStaking = 0,
-        Assembly = 1,
-        Shimmer = 2,
-        AssemblyAndShimmer = 3,
-    }
-
     function setAnimation(): void {
-        const prefix = 'staking-info'
-        if (!stakingEventState || !$selectedAccountParticipationOverview) {
-            animation = `${prefix}-upcoming`
+        if (!$selectedAccountParticipationOverview) {
+            animation = StakingAnimation.Neither
         }
 
-        if (stakingEventState === ParticipationEventState.Inactive) {
-            animation = null
-        } else if (stakingEventState === ParticipationEventState.Holding) {
-            let fileNumber = AnimationFileNumber.NoStaking
-            if (isAssemblyStaked && isShimmerStaked) {
-                fileNumber = AnimationFileNumber.AssemblyAndShimmer
-            } else if (isAssemblyStaked) {
-                fileNumber = AnimationFileNumber.Assembly
-            } else if (isShimmerStaked) {
-                fileNumber = AnimationFileNumber.Shimmer
+        switch (stakingEventState) {
+            case ParticipationEventState.Upcoming:
+            case ParticipationEventState.Commencing:
+                animation = StakingAnimation.Prestaking
+                break
+            case ParticipationEventState.Holding: {
+                const participatingEventIds = $selectedAccountParticipationOverview.participations.map((p) => p.eventId)
+
+                const isStakingForAssembly = participatingEventIds.includes(ASSEMBLY_EVENT_ID)
+                const isStakingForShimmer = participatingEventIds.includes(SHIMMER_EVENT_ID)
+
+                if (isStakingForAssembly && isStakingForShimmer) {
+                    animation = StakingAnimation.Both
+                } else if (!isStakingForAssembly && !isStakingForShimmer) {
+                    animation = StakingAnimation.Neither
+                } else {
+                    if (isStakingForAssembly) {
+                        const hasShimmerRewards = $selectedAccountParticipationOverview.shimmerRewards > 0
+                        animation = hasShimmerRewards
+                            ? StakingAnimation.AssemblyWithShimmerRewards
+                            : StakingAnimation.AssemblyWithoutShimmerRewards
+                    } else if (isStakingForShimmer) {
+                        const hasAssemblyRewards = $selectedAccountParticipationOverview.assemblyRewards > 0
+                        animation = hasAssemblyRewards
+                            ? StakingAnimation.ShimmerWithAssemblyRewards
+                            : StakingAnimation.ShimmerWithoutAssemblyRewards
+                    } else {
+                        animation = StakingAnimation.Neither
```

Comment:
I want to clarify that this statement will never be reached in this code (covered by [line 66](https://github.com/iotaledger/firefly/pull/2885/files#diff-2d43240046c8f964430ba955efabf3823349ee67637349294f978bb6bdec0776R66)), but I wanted to add it just for completeness.